### ISSUE TITLE
Don't campaign anymore if quorum rejects vote for log gap

### ISSUE THREAD

**Issue by BusyJay** [pingcap/raft-rs#17]
Apparently, if a peer's votes get rejected by quorum for log gap, then the peer can't become leader anymore by simply retry. So we can disable campaign for these peers until a leader is elected.

Maybe related to #14.

**Comment by siddontang** [comment]
PreVote can fix this problem?

**Comment by BusyJay** [comment]
Yes. PreVote is an optional configuration, which should not be counted on.
